app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlAdapter.kt

```diff
@@ -245,11 +247,15 @@ class AdapterItemDiffCallback : DiffUtil.ItemCallback<AdapterItem>() {
     }
 }
 
+@Suppress("LongParameterList")
 class SessionControlAdapter(
     private val store: HomeFragmentStore,
     private val interactor: SessionControlInteractor,
     private val viewLifecycleOwner: LifecycleOwner,
-    private val components: Components
+    private val components: Components,
+    private val onPocketStoriesShown: (List<PocketRecommendedStory>) -> Unit,
+    private val onCategoryClick: (PocketRecommendedStoryCategory) -> Unit,
+    private val onExternalLinkTapped: (String) -> Unit
```

Comment:
Can we add this to an Interactor for Pocket?

Reply:
Talked about this with Sebastian some time ago, where should all the logic stay when using composables.
Things being new we have to try a few approaches to find what works but the general plans would be to try and move away from Interactors/Controllers. For the start. Then continue moving away from even more of the current architecture. More thoughts here - https://docs.google.com/document/d/1BG_aGO_dCP7c1ViY3UG3Rqqi6PCJlRLDfI3LVluWyGM/edit# .

Reply:
I have read the document and understand where the proposals are coming from. Bigger picture to me would be that everything in SessionControlAdapter is completely remove since everything would be converted to Jetpack Compose in the future.

What I don't particularly want to see is that this becomes an intermediate example that people follow and continue to overload SessionControlAdapter with additional parameters as we can also assume this will go away in the future. As an example, where would we draw the line now if we add more compose components and keep adding onRecentTabClick, onRecentBookmarkClick, onRecentVisitClick into this param list. I think we should keep to our existing lint rule on not running a big parameter list. 

The cleaner solution even though we are sticking with encapsulating common logic into Interactor still holds valid here.

Reply:
@pocmo ?

Reply:
Another thought is that we still want to be able to write tests for the business logic that might be called by `onPocketStoriesShown`, etc. It is still beneficial to throw the functionality within a controller-type class. In general, I am in favour of dropping one step of indirections caused by the Interactors.

Reply:
Added an Interactor and Controller in https://github.com/mozilla-mobile/fenix/pull/21104 to be used for all later functionality incl this.